src/main/java/data/Database.java

```diff
@@ -55,25 +60,25 @@ public OutputStream writeData(String runId, String type) {
     return null;
   }
 
+  /*Returns appropiate file name for data with this runId and type. */
   private String makeFileName(String runId, String type) {
-    return getDatabase() + "/" + runId + "_" + type;
+    return getDirectoryName() + "/" + runId + "_" + type;
   }
 
   /* Adds file to database given its runid and type. */
   public String newDatabaseEntry(String runId, String type) {
     String fileName = makeFileName(runId, type);
-    files.add(fileName);
     return fileName;
   }
 
   /* Return name of file with runid and type in database. */
   public String findName(String runId, String type) {
-    String fileName = getDatabase() + "/" + runId + "_" + type;
-    return files.get(0);
+    return makeFileName(runId, type);
   }
 
   /* Returns list of files in database. */
-  public ArrayList<String> getAllFiles() {
-    return files;
+  public List<File> getAllFiles() {
+    File[] files = getDatabase().listFiles();
+    return Arrays.asList(files);
```

Comment:
instead of keeping and updating my own list of files added to the directory, i call Arrays.asList, which is more accurate and efficient than my original method.

Reply:
What's the motivation for storing the files object, the database will have to return a list of all its entries as one of its methods, which makes sense to return as a string? I imagine there is additional overhead for storing all these file objects, could you explain what benefit there would be. If the conclusion if I want to open a file I already have the object I don't think that makes more sense than just creating the file object when you need it?